File: forge/src/main/java/net/shiroha233/roadweaver/features/forge/RoadFeaturesForge.java
package net.shiroha233.roadweaver.features.forge;

import net.minecraft.core.registries.Registries;
import net.minecraft.world.level.levelgen.feature.Feature;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.RegistryObject;
import net.shiroha233.roadweaver.RoadWeaver;
import net.shiroha233.roadweaver.features.path.config.PathFeatureConfig;
import net.shiroha233.roadweaver.features.path.PathFeature;

public final class RoadFeaturesForge {
    private RoadFeaturesForge() {}

    public static final DeferredRegister<Feature<?>> FEATURES =
            DeferredRegister.create(Registries.FEATURE, RoadWeaver.MOD_ID);

    public static final RegistryObject<Feature<PathFeatureConfig>> ROAD_FEATURE = FEATURES.register(
            "road_feature",
            () -> new PathFeature(PathFeatureConfig.CODEC)
    );

    public static void register(IEventBus modBus) {
        FEATURES.register(modBus);
    }
}

File: forge/src/main/kotlin/net/shiroha233/roadweaver/features/forge/RoadFeaturesForge.kt
package net.shiroha233.roadweaver.features.forge

import net.minecraft.core.registries.Registries
import net.minecraft.world.level.levelgen.feature.Feature
import net.minecraftforge.eventbus.api.IEventBus
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.RegistryObject
import net.shiroha233.roadweaver.RoadWeaver
import net.shiroha233.roadweaver.features.path.PathFeature
import net.shiroha233.roadweaver.features.path.config.PathFeatureConfig
import java.util.function.Supplier

class RoadFeaturesForge private constructor() {
    companion object {
        @JvmField
        val FEATURES: DeferredRegister<Feature<*>> = DeferredRegister.create(Registries.FEATURE, RoadWeaver.MOD_ID)

        @JvmField
        val ROAD_FEATURE: RegistryObject<Feature<PathFeatureConfig>> = FEATURES.register("road_feature") {
            PathFeature(PathFeatureConfig.CODEC)
        }

        @JvmStatic
        fun register(modBus: IEventBus) {
            FEATURES.register(modBus)
        }
    }
}
